doc/smart-switch/PA-Validation/SmartSwitchPAValidationOffload.md

```diff
@@ -0,0 +1,172 @@
+- DashOffloadManager should listen to the DASH configuration and either intercept it for offloading, or forward it to the DPU.
+- Provide PA validation support in NPU when this feature is not supported in DPU pipeline model.
+- Use existing ACL tables and ACL rules design in NPU.
+
+## 1.2 Scale Requirements
+PA Validation configuration:
+* 4096 PA Validation entries
+
+Please refer to [SONiC-DASH HLD](https://github.com/sonic-net/SONiC/blob/master/doc/dash/dash-sonic-hld.md#3213-underlay-src-ip-pa-validation) for more details regarding the PA Validation configuration requirements
+
+ACL:
+- One ACL table per each DPU
+- 4160 ACL rules:
+  - 4096 ACL forward rules (one per each PA Validation address)
+  - 64 ACL drop rules (one per each unique VNI in the configuration)
+
+# 2 Modules Design
+
+## 2.1 Dash Offload Manager
+The new orchagent application DashOffloadManager will be responsible for DASH offloading logic. It will collect all the needed information for offloading and perform all the relevant configurations.
+To get the DASH configuration that should be offloaded, the DashOffloadManager will act as a transparent ZMQ proxy between the GNMI server and the DPU swss, forwarding all the configuration and intercepting the tables that should be offloaded.
+To simplify the management of the DPU offload and achieve optimal performance, every DPU is handled by a separate instance of a ZMQ Proxy (pair of ZMQ Server and Client)
+
+<img src="images/DashOffloadManager.svg">
+
+Once the offload is required, the DashOffloadManager will start designated orch (e.g. PAValidationOffloadOrch) that will subscribe to the configuration and do the offload logic.
+
+<img src="images/DashOffloadManagerWithConsumer.svg">
+
+The Dash Offload Manager is disabled by default and only enabled for specific platforms that require its functionality.
+
+### 2.1.1 SONiC GNMI server changes
+A new parameter `zmq_dpu_proxy_address_base` is added to the telemetry.go to enable the GNMI Server -> DASH Offload Manager ZMQ connection. If specified, the GNMI server will use `zmq_dpu_proxy_address_base + dpu_index:zmq_port` as a destination for DPU ZMQ connection. For example, on the system with 2 DPUs and base address 127.0.0.10, the DASH Offload Manager's ZMQ server runs on 127.0.0.10:8100 for DPU0 and on 127.0.0.11:8100 for DPU1.
+
+## 2.2 PA Validation Offload
+
+A new orchagent (PaValidationOffloadOrch) is responsible for offloading the PA_VALIDATION table entries. Once started by DashOffloadManager, it subscribes to the PA_VALIDATION table and maps the PA validation entries into the ACL configuration.
+
+### 2.2.1 PA Validation Offload ACL configuration
+
+The offloading is achieved by using a User-Defined EGRESS ACL table bound to the NPU<->DPU port. The table is created/managed per DPU.
+
+```
+{
+    "ACL_TABLE_TYPE": {
+        "DASH_PA_VALIDATION": {
+            "MATCHES": [
+                "TUNNEL_VNI",
+                "SRC_IP",
+                "SRC_IPV6"

```

Comment:
it will be better to add the destination IP in here as well. The reason is because SmartSwitch lives in T1 and can receive other traffic in the same VNET, which is not sent to the DPU, but to other VMs. Adding the SmartSwitch data plane VIP as destination IP will be safer and more future-proof.

Reply:
@r12f These ACLs are attached to DPU connected ports in the Egress direction. So, forwarding rule(based on DPU VIP route or ENI-ACL based forwarding) makes sure that only DPU destined traffic are egressing NPU towards DPU. So, dest_ip is not required. It will also increase the TCAM key width and may impact the scale numbers for this ACL and also other T1 specific Egress ACLs.

Reply:
Got it! Egress ACL should work!